opm/simulators/flow/OutputBlackoilModule.hpp

```diff
@@ -1146,12 +1146,18 @@ class OutputBlackOilModule : public GenericOutputBlackoilModule<GetPropType<Type
             MaterialLaw::capillaryPressures(pc, matParams, fs);
             Valgrind::CheckDefined(this->fluidPressure_[elemIdx]);
             Valgrind::CheckDefined(pc);
-            assert(FluidSystem::phaseIsActive(oilPhaseIdx));
+            const auto& pressure = this->fluidPressure_[elemIdx];
             for (unsigned phaseIdx = 0; phaseIdx < numPhases; ++phaseIdx) {
                 if (!FluidSystem::phaseIsActive(phaseIdx))
                     continue;
 
-                fs.setPressure(phaseIdx, this->fluidPressure_[elemIdx] + (pc[phaseIdx] - pc[oilPhaseIdx]));
+                if (Indices::oilEnabled)
+                    fs.setPressure(phaseIdx, pressure + (pc[phaseIdx] - pc[oilPhaseIdx]));
+                else if (Indices::gasEnabled)
+                    fs.setPressure(phaseIdx, pressure + (pc[phaseIdx] - pc[gasPhaseIdx]));
```

Comment:
Did you verify that `pressure + (...)` matches the sign definition of the capillary pressure provided by `MaterialLaw::capillaryPressure()`?

Reply:
Yes, as I understand the code, this is correct. 

This also aligns with what is done in https://github.com/OPM/opm-simulators/blob/c7b3e40bfae7db8fb964465bc89dc334d5f9ebfd/opm/simulators/flow/FlowProblemBlackoil.hpp#L1328